app/metrics/views/common.py

```diff
@@ -114,13 +116,37 @@ def generate_tables_and_figures(group, values):
 
 def get_callback(group):
     def update_graph_and_table(*filter_values):
-        filters = list(zip([*group.get_filter_fields(), "date_from", "date_to", "node_only"], filter_values))
+        filters = list(zip([*group.get_filter_fields(), "date_from", "date_to", "node_only"], filter_values[:-len(group.get_fields())]))
+        chart_types = filter_values[-len(group.get_fields()):]  # Last inputs are the chart types
         values = group.get_values(**{
             name: value
             for name, value in filters
         })
+        outputs = []
+        for field_id, chart_type in zip(group.get_fields(), chart_types):
```

Comment:
Maybe you could make sure to remove the function `generate_tables_and_figures` too since it is no longer in use.

Reply:
Removed the function